lib/spack/external/archspec/json/cpu/microarchitectures.json

```diff
@@ -262,6 +290,20 @@
             "name": "x86-64",
             "flags": "-march={name} -mtune=generic -mcx16 -msahf -mpopcnt -msse3 -msse4.1 -msse4.2 -mssse3 -mavx -mavx2 -mbmi -mbmi2 -mf16c -mfma -mlzcnt -mmovbe -mxsave -mavx512f -mavx512bw -mavx512cd -mavx512dq -mavx512vl"
           }
+        ],
+        "oneapi": [
+          {
+            "versions": ":",
+            "name": "x86-64-v4",
```

Comment:
Where are these coming from? I do not see x86-64-v[2-4] in the manual pages or in strings output.

Reply:
Okay; it looks like v2022 has those but v2021 does not. They are not in the manual page for either but they can be seen with strings output.

Reply:
Okay, I figured out how to make the compiler print out the targets and determined that version 2021.2.0 introduced support for x86-64-v[2-4]. I submitted https://github.com/archspec/archspec-json/pull/49 to indicate that.

Reply:
@glennpj Apologies for the lack of context, this was done in a rush to see if we could have it in v0.18.0. I have been reported this error on Slack: https://github.com/rscohn2/oneapi-spack-tests/runs/6597798299?check_suite_focus=true The error seemed to be due to `archspec` having "holes" in compiler support for `oneapi`. Specifically it was missing generic `x86_64` targets.

While I intend to investigate the root of the issue better after ISC, this was meant to be both an hotfix for that pipeline and an addition to the JSON (since we want to have flags for the generic architectures). I'll be reviewing your PR in `archspec` right away, thanks!

Reply:
@rscohn2  @glennpj @becker33 The PR has been updated.